WalletWasabi.Gui/ViewModels/AddressViewModel.cs

```diff
@@ -134,10 +138,40 @@ public bool InEditMode
 			set => this.RaiseAndSetIfChanged(ref _qrCode, value);
 		}
 
+		public string QRImageSavePath
+		{
+			get => _qRImageSavePath;
+			set => this.RaiseAndSetIfChanged(ref _qRImageSavePath, value);
+		}
+
 		public string ExpandMenuCaption => _expandMenuCaption?.Value ?? string.Empty;
 
 		public CancellationTokenSource CancelClipboardNotification { get; set; }
 
+		public async Task SaveQRCodeAsync()
+		{
+			var sfd = new SaveFileDialog();
+
+			sfd.InitialFileName = $"{Address}.png";
+			sfd.Directory = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
+			sfd.Filters.Add(new FileDialogFilter() { Name = "Portable Network Graphics (PNG) Image file", Extensions = { "png" } });
+
+			var visualRoot = (ClassicDesktopStyleApplicationLifetime)Application.Current.ApplicationLifetime;
+			var path = await sfd.ShowAsync(visualRoot.MainWindow);
+
+			if (!string.IsNullOrWhiteSpace(path))
+			{
+				var ext = Path.GetExtension(path);
+
+				if (string.IsNullOrWhiteSpace(ext) || ext.ToLowerInvariant() != "png")
+				{
+					path = $"{path}.png";
+				}
+				
+				QRImageSavePath = path;
```

Comment:
I would do the job here. Open the file dialog and save, without the use of Rx property notify chain.

Reply:
That might be a violation of MVVM because Dialogs stuff are view/framework side code im afraid...

Reply:
You are right because QrCode is a view element. Could you replace the current logic with a SaveToFileCommand, and call it instead of setting QRImageSavePath. Set a property to save a file is an unexpected behavior for me.